tests/neptune/new/sync/utils.py

```diff
@@ -0,0 +1,140 @@
+    def get_run_impl(run_id):
+        """This function will return run as well as projects. Will be cleaned in ModelRegistry"""
+        for exp in registered_experiments:
+            if run_id in (str(exp.id), get_qualified_name(exp)):
+                return exp
+
+    return get_run_impl
+
+
+def prepare_projects(path):
+    unsync_project = api_project()
+    sync_project = api_project()
+    registered_projects = (unsync_project, sync_project)
+
+    execution_id = "exec-0"
+
+    for project in registered_projects:
+        project_path = path / "async" / str(project.id) / execution_id
+        project_path.mkdir(parents=True)
+        queue = DiskQueue(
+            dir_path=project_path,
+            to_dict=lambda x: x,
+            from_dict=lambda x: x,
+            lock=threading.RLock(),
+        )
+        queue.put("op-proj-0")
+        queue.put("op-proj-1")
+
+    SyncOffsetFile(
+        path / "async" / str(unsync_project.id) / execution_id / "last_ack_version"
+    ).write(1)
+    SyncOffsetFile(
+        path / "async" / str(unsync_project.id) / execution_id / "last_put_version"
+    ).write(2)
+
+    SyncOffsetFile(
+        path / "async" / str(sync_project.id) / execution_id / "last_ack_version"
+    ).write(2)
+    SyncOffsetFile(
+        path / "async" / str(sync_project.id) / execution_id / "last_put_version"
+    ).write(2)
+
+    return unsync_project, sync_project, generate_get_run_impl(registered_projects)
+
+
+def prepare_runs(path):
+    unsync_exp = api_run()
+    sync_exp = api_run()
+    registered_runs = (unsync_exp, sync_exp)
+
+    execution_id = "exec-0"
+
+    for exp in registered_runs:
+        exp_path = path / "async" / f"{exp.type.value}__{exp.id}" / execution_id
+        exp_path.mkdir(parents=True)
+        queue = DiskQueue(
+            dir_path=exp_path,
+            to_dict=lambda x: x,
+            from_dict=lambda x: x,
+            lock=threading.RLock(),
+        )
+        queue.put("op-0")
+        queue.put("op-1")
+
+    SyncOffsetFile(
+        path
+        / "async"
+        / f"{unsync_exp.type.value}__{unsync_exp.id}"
+        / execution_id
+        / "last_ack_version"
+    ).write(1)
+    SyncOffsetFile(
+        path
+        / "async"
+        / f"{unsync_exp.type.value}__{unsync_exp.id}"
+        / execution_id
+        / "last_put_version"
+    ).write(2)
+
+    SyncOffsetFile(
+        path
+        / "async"
+        / f"{sync_exp.type.value}__{sync_exp.id}"
+        / execution_id
+        / "last_ack_version"
+    ).write(2)
+    SyncOffsetFile(
+        path
+        / "async"
+        / f"{sync_exp.type.value}__{sync_exp.id}"
+        / execution_id
+        / "last_put_version"
+    ).write(2)
+
+    return unsync_exp, sync_exp, generate_get_run_impl(registered_runs)
+
+
+def prepare_offline_run(path):
+    offline_exp_uuid = str(uuid.uuid4())
+    offline_exp_path = path / OFFLINE_DIRECTORY / f"run__{offline_exp_uuid}"
+    offline_exp_path.mkdir(parents=True)
```

Comment:
please add another run with "old" storage path (`offline/{exp._id}`)

Reply:
Tests were refactored and there are separate cases for deprecated RUN format